### PR TITLE
NXP-28139: add docker and docker-json templates

### PR DESCRIPTION
**Pull request by NourNuxeo**
The goal is to stop logging to files when using the docker template and to log in json when using the docker-json template. So the newly introduced log4j2.xml are similar to the original Tomcat `/server/nuxeo-server-tomcat/src/main/resources/tomcat/lib/log4j2.xml` file but they have no file appenders.

Please note there are still log files used by the nuxeo Launcher project which has its own embeded log4j2.xml configuration. It logs limited amount of infos concerning Launch and ConfigGenerator (the launching and stopping infos, the included templates, time taken to start and stop event..)
That log4j2.xml file is embeded in the final nuxeo-launcher-*.jar so wen can't change it easily.

### CODE REVIEW

File: server/nuxeo-nxr-server/src/main/resources/templates/docker-json/lib/log4j2.xml [hunk 1]
```diff
@@ -0,0 +1,208 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Log4j Configuration - For more configuration information and examples see https://logging.apache.org/log4j/2.x/ -->
+<Configuration monitorInterval="30">
+  <Appenders>
+    <!-- ============================== -->
+    <!-- Append messages to the console -->
+    <!-- ============================== -->
+    <Console name="ConsoleJSONAppender" target="SYSTEM_OUT">
+        <JsonLayout complete="false" compact="false"/>
```
Comment (kevinleturc) [inline server/nuxeo-nxr-server/src/main/resources/templates/docker-json/lib/log4j2.xml:9]: We should enable `compact` for a production configuration which is the intent of this file.

Furthemore, I think `eventEol` should be enable too in order to have one log per line.

`complete` looks like a debug option to me and shouldn't be changed:
> If true, the appender includes the JSON header and footer, and comma between records. Defaults to false.

File: server/nuxeo-nxr-server/src/main/resources/templates/docker-json/lib/log4j2.xml [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    <Logger name="org.nuxeo.elasticsearch" level="trace" additivity="false">
+      <AppenderRef ref="ELASTIC" />
+    </Logger>
+    -->
+
+    <!-- =================== -->
+    <!-- Third Party Loggers -->
+    <!-- =================== -->
+    <Logger name="org.apache.catalina.loader" level="warn" additivity="false">
+      <AppenderRef ref="CLASSLOADER" />
```
Comment (kevinleturc) [inline server/nuxeo-nxr-server/src/main/resources/templates/docker-json/lib/log4j2.xml:138]: This appender is not defined.

File: server/nuxeo-nxr-server/src/main/resources/templates/docker-json/lib/log4j2.xml [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+    </Logger>
+    <Logger name="org.apache.catalina.core.AprLifecycleListener" level="info" additivity="false">
+      <AppenderRef ref="TOMCAT" />
```
Comment (kevinleturc) [inline server/nuxeo-nxr-server/src/main/resources/templates/docker-json/lib/log4j2.xml:141]: This appender is not defined.

File: server/nuxeo-nxr-server/src/main/resources/templates/docker/lib/log4j2.xml [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+    <Logger name="org.nuxeo.elasticsearch" level="trace" additivity="false">
+      <AppenderRef ref="ELASTIC" />
+    </Logger>
+    -->
+
+    <!-- =================== -->
+    <!-- Third Party Loggers -->
+    <!-- =================== -->
+    <Logger name="org.apache.catalina.loader" level="warn" additivity="false">
+      <AppenderRef ref="CLASSLOADER" />
```
Comment (kevinleturc) [inline server/nuxeo-nxr-server/src/main/resources/templates/docker/lib/log4j2.xml:139]: Appender not defined

File: server/nuxeo-nxr-server/src/main/resources/templates/docker/lib/log4j2.xml [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+    </Logger>
+    <Logger name="org.apache.catalina.core.AprLifecycleListener" level="info" additivity="false">
+      <AppenderRef ref="TOMCAT" />
```
Comment (kevinleturc) [inline server/nuxeo-nxr-server/src/main/resources/templates/docker/lib/log4j2.xml:142]: Appender not defined.

File: server/nuxeo-server-tomcat/src/main/assemble/assembly.xml [hunk 6]
```diff
@@ -180,6 +180,8 @@
         <include name="log4j-*" /> <!-- log4j2 / bridges: jcl, slf4j, log4j -->
         <include name="json-smart-*" />
         <include name="jsonevent-layout-*" />
+        <!-- install log4j2 JSON logging dependencies in container classpath -->
+        <include name="jackson-*" />
```
Comment (kevinleturc) [inline server/nuxeo-server-tomcat/src/main/assemble/assembly.xml:184]: The two lines preceding the one you introduced also deals with JSON and was added for [NXP-25096](https://jira.nuxeo.com/browse/NXP-25096).

The ticket is mainly about log4j 1 so we should be able to remove these lines.
Note: Link the given ticket to yours as it the same topic just a different logging backend.

Comment (NourNuxeo) [reply server/nuxeo-server-tomcat/src/main/assemble/assembly.xml:182]: I added it as related to this issue explaining it was config for log4j1

File: server/nuxeo-nxr-server/src/main/resources/templates/docker-json/lib/log4j2.xml [hunk 7]
```diff
@@ -0,0 +1,201 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Log4j Configuration - For more configuration information and examples see https://logging.apache.org/log4j/2.x/ -->
+<Configuration monitorInterval="30">
+  <Appenders>
+    <!-- ============================== -->
+    <!-- Append messages to the console -->
+    <!-- ============================== -->
+    <Console name="ConsoleJSONAppender" target="SYSTEM_OUT">
+        <JsonLayout eventEol="true" compact="true"/>
```
Comment (troger) [inline server/nuxeo-nxr-server/src/main/resources/templates/docker-json/lib/log4j2.xml:9]:
```suggestion
      <JsonLayout eventEol="true" compact="true"/>
```

**Pull request merged**